docs/source/testing.rst

```diff
@@ -0,0 +1,868 @@
+Testing
+==========
+
+
+Let's take a look at how 🤗 Transformer models are tested and how you can write new tests and improve the existing ones.
+
+There are 2 test suites in the repository:
+
+1. ``tests`` -- tests for the general API
+2. ``examples`` -- tests primarily for various applications that aren't part of the API
+
+How transformers are tested
+---------------------------
+
+1. Once a PR is submitted it gets tested with 9 CircleCi jobs. Every new commit to that PR gets retested. These jobs are defined in this `config file <https://github.com/huggingface/transformers/blob/master/.circleci/config.yml>`__, so that if needed you can reproduce the same environment on your machine.
+   
+   These CI jobs don't run ``@slow`` tests.
+   
+2. There are 3 jobs run by `github actions <https://github.com/huggingface/transformers/actions>`__:
+
+   * `torch hub integration <https://github.com/huggingface/transformers/blob/master/.github/workflows/github-torch-hub.yml>`__:  checks whether torch hub integration works. Julien is in charge if that breaks. It doesn’t break very often.
+
+   * `self-hosted (push) <https://github.com/huggingface/transformers/blob/master/.github/workflows/self-push.yml>`__: runs fast tests on GPU every PR.
```

Comment:
No, that runs only on commits on `master`, not on every PR. It only runs if a commit on `master` has updated the code in one of the following folders: `src/**`, `tests/**`, `.github/**` (to prevent running on added model cards, notebooks etc)

Reply:
Thank you for that clarification, @LysandreJik.